docs/src/components/action/NpmCopyButton.tsx

```diff
@@ -1,63 +1,60 @@
 import * as React from 'react';
 import copy from 'clipboard-copy';
-import Button, { ButtonProps } from '@mui/base/Button';
-import { styled, SxProps } from '@mui/system';
-import { Theme } from '@mui/material/styles';
+import ButtonUnstyled, { ButtonProps } from '@mui/base/Button';
+import { styled, SxProps, Theme } from '@mui/system';
 import ContentCopyRounded from '@mui/icons-material/ContentCopyRounded';
 import CheckRounded from '@mui/icons-material/CheckRounded';
 
-const StyledButton = styled(Button)`
-  display: inline-flex;
-  gap: 8px;
-  align-items: center;
-  justify-content: center;
-  position: relative;
-  box-sizing: border-box;
-  -webkit-tap-highlight-color: transparent;
-  outline: 0;
-  border: 0;
-  margin: 0;
-  vertical-align: middle;
-  text-decoration: none;
-  text-transform: initial;
-  min-width: 64px;
-  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
-    color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
-  box-shadow: none;
-  cursor: copy;
-  padding: 0;
-  width: max-content;
-  background-color: transparent;
-  color: var(--muidocs-palette-grey-600);
-  font-family: Menlo, Consolas, 'Droid Sans Mono', monospace;
-  font-weight: 400;
-  font-size: 0.75rem;
-  line-height: 1.5;
-  letter-spacing: 0;
-  -webkit-font-smoothing: subpixel-antialiased;
-  margin-top: 16px;
+const Button = styled(ButtonUnstyled)(({ theme }) => ({
+  boxSizing: 'border-box',
+  display: 'inline-flex',
+  gap: 8,
+  alignItems: 'center',
+  justifyContent: 'center',
+  position: 'relative',
+  '-webkit-tap-highlight-color': 'transparent',
+  outline: 0,
+  border: 0,
+  margin: 0,
+  verticalAlign: 'middle',
+  textDecoration: 'none',
+  textTransform: 'initial',
+  minWidth: 64,
+  // transition: 'background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms','color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+  boxShadow: 'none',
+  cursor: 'copy',
+  padding: 0,
+  width: 'max-content',
+  backgroundColor: 'transparent',
+  color: theme.palette.grey[600],
+  // fontFamily: theme.typography.fontFamilyCode,
+  // fontSize: theme.typography.pxToRem(14),
```

Comment:
I'm having a hard time converting the styles of this button to CSS in JS 😅 

Initially, I was trying this pattern:`(theme.vars || theme).palette.grey[600],`, for which I was being thrown an error of "`vars` is not assigned on the Theme". Alright, so let's ditch `vars` and just have `theme` there. For colors, it worked fine but... for the below font-related properties, I'm still seeing an error that doesn't make sense to me:

<img width="250" alt="Screen Shot 2023-08-03 at 07 39 03" src="https://github.com/mui/material-ui/assets/67129314/399fc874-4ba8-44a8-b32a-827adc52b1b9">

I mean... they _are_ available on the theme, lots of other components use them... 
What am I doing wrong? 🤔 All the commented styles here will be put back once this is figured out!

Reply:
This is because `styled` is imported from `@mui/system`. You can change the import to:

```js
import { styled } from '@mui/material/styles';
```

Then the theme type should have `vars`.

Reply:
> I'm having a hard time converting the styles of this button to CSS in JS 😅

Try https://transform.tools/css-to-js

Reply:
Oh, sweet, thanks Jun! Everything seems to be working out now 🤙

Reply:
@siriwatknp Nice, thanks for helping out!